### ISSUE TITLE
Style/GuardClause configurable list of methods that are guards

### ISSUE THREAD

**Issue by TSMMark** [rubocop/rubocop#11761]
## Is your feature request related to a problem? Please describe.

Style/GuardClause is described as such:

```ruby
# A condition with an `elsif` or `else` branch is allowed unless
# one of `return`, `break`, `next`, `raise`, or `fail` is used
# in the body of the conditional expression.
```

[A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]](https://github.com/rubocop/rubocop/blob/7c35167233e567cd0bad3879974de787b3c372bd/lib/rubocop/cop/style/guard_clause.rb#L9-L11)

However, it's common for libraries, gems, application codebases, to introduce methods which are guards, and will `raise` internally. I would like to be able to configure a list of methods (in addition to the defaults), that should be considered as guard clause methods.

For example, in Roda gem, the method `r.redirect` method https://github.com/jeremyevans/roda#usage-

`r.redirect` immediately returns the response,
allowing for code such as `r.redirect(path) if some_condition`

Cuba gem has similar `res.redirect "/hello"` method which also raises internally and behaves the same way.

I'm sure there are other libraries with similar methods.

Style/GuardClause rule could handle this:

```
# bad
if whatever?
  r.redirect("/path")
else
  render_page
end
```

```
# good
r.redirect("/path") if whatever?

render_page
```

## Describe the solution you'd like

Configurable list of additional methods that will be considered guards. Something like:

```
Style/GuardClause:
  Enabled: true
  GuardMethods:
    - 'r.redirect'
    - 'res.redirect'
```

---

I may be able to implement this. Let me know if such a change could be accepted. Thanks!

**Comment by TSMMark** [comment]
I think it's a good idea and I stand by

**Comment by TSMMark** [comment]
Would maintainers accept a PR for this?
